gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformHintRule.scala

```diff
@@ -421,7 +395,7 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
               plan.leftKeys,
               plan.rightKeys,
               plan.joinType,
-              TransformHints.getShuffleHashJoinBuildSide(plan),
```

Comment:
Do we need change it? The build side may be changed when we offload shj, and we should use the target build side to do validate.

Reply:
There is no problem with the current logic, but there may be hidden dangers in future logic changes. I will modify it immediately.

Reply:
I thought about it carefully. The buildSide itself is the legal buildSide in the corresponding backends obtained. The buildSide will not be verified in doValidate, and it should not be done in the future. If there are changes in the future, the corresponding supportHashBuildJoinTypeOnLeft and supportHashBuildJoinTypeOnRight should be changed.

Reply:
The buildSide here is from vanilla Spark rather than Gluten backends.

Reply:
yea, but I think there is no need to convert here, because do validate does not require buildside.

Reply:
We should still stay consistent to avoid risks, updated.